bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/api/EcobeeApi.java

```diff
@@ -344,11 +346,13 @@ private boolean isSuccess(@Nullable AbstractResponseDTO response) {
                 createOAuthClientService();
             } else if (response.status.code == ECOBEE_TOKEN_EXPIRED) {
                 // Check isAuthorized again to see if we can get a valid token
-                logger.debug("API: Unable to complete API call because token is expired");
+                logger.info("API: Unable to complete API call because token is expired. Try to refresh the token...");
+                closeOAuthClientService();
+                createOAuthClientService();
```

Comment:
The other comments aside, assuming that a work-around will in fact be needed, do you think this would be enough (and more light-weight)?
```suggestion
                oAuthClientService.refreshToken()
```

Reply:
Yes, good idea. I think that will work.

Reply:
I think you will then need remove lines 356-358 to not do both. 😉 Additionally, after triggering the token refresh, you should probably:
```java
return isAuthorized();
```

or similar. `refreshToken()` returns the new token, but `isAuthorized()` will store it, validate it and use it, I think.